### ISSUE TITLE
Poloniex WSS client forces autobahn into to use txaio asyncio

### ISSUE THREAD

**Issue by emestee** [Crypto-toolbox/bitex#148]
Since autobahn's underlying framework selection is stateful, any other code that also uses autobahn and requests twisted will fail. Conversely, if any other code has called `txaio.use_twisted()` before Poloniex initialization (which is triggered by loading of `__init__.py` in bitex.api.WSS, meaning _any_ use of bitex will cause it, even if Poloniex integration isn't required). For instance:

```python
import txaio
txaio.use_twisted()
from bitex.api.REST import HitBTCREST
```
produces the following:

```
Traceback (most recent call last):
  File "test.py", line 4, in <module>
    from bitex.api.REST import HitBTCREST
  File "/home/kek/dev/python/.env/lib/python3.6/site-packages/bitex/__init__.py", line 5, in <module>
    from bitex.interfaces import Kraken, Bitfinex, Bitstamp, CCEX, Coincheck
  File "/home/kek/dev/python/.env/lib/python3.6/site-packages/bitex/interfaces/__init__.py", line 1, in <module>
    from bitex.interfaces.bitfinex import Bitfinex
  File "/home/kek/dev/python/.env/lib/python3.6/site-packages/bitex/interfaces/bitfinex.py", line 11, in <module>
    from bitex.api.REST import BitfinexREST
  File "/home/kek/dev/python/.env/lib/python3.6/site-packages/bitex/api/__init__.py", line 2, in <module>
    import bitex.api.WSS
  File "/home/kek/dev/python/.env/lib/python3.6/site-packages/bitex/api/WSS/__init__.py", line 7, in <module>
    from bitex.api.WSS.poloniex import PoloniexWSS
  File "/home/kek/dev/python/.env/lib/python3.6/site-packages/bitex/api/WSS/poloniex.py", line 8, in <module>
    from autobahn.asyncio.wamp import ApplicationRunner, ApplicationSession
  File "/home/kek/dev/python/.env/lib/python3.6/site-packages/autobahn/asyncio/__init__.py", line 36, in <module>
    from autobahn.asyncio.websocket import \
  File "/home/kek/dev/python/.env/lib/python3.6/site-packages/autobahn/asyncio/websocket.py", line 32, in <module>
    txaio.use_asyncio()
  File "/home/kek/dev/python/.env/lib/python3.6/site-packages/txaio/__init__.py", line 120, in use_asyncio
    raise RuntimeError("Explicitly using '{}' already".format(_explicit_framework))
RuntimeError: Explicitly using 'twisted' already
```

Arguably, this is a design flaw in autobahn & txaio, but it does affect the dependent code in bitex. As a concrete instance of a problem, the `python-binance` library can not be used in the same execution context as bitex

**Comment by nlsdfnbch** [comment]
Since this is a websocket issue, and websockets are deprecated, I won't fix this. The next release of `BitEx` won't feature them so this issue should take care of itself with the next release.

**Comment by andy-io** [comment]
@emestee did you find any solution for this ? i have the same problem ...

**Comment by emestee** [comment]
@andy-io the only solution is to either change the txaio engine, which is hardcoded in bitex, or disable the Poloniex integration. In any case you will end up editing bitex code, so fork it and install it from your copy.

**Comment by andy-io** [comment]
@emestee thanks for your reply...could you show me which part of the code in bitex must be changed exactly ? i dont have any idea about autobahn so could you please help me...

**Comment by emestee** [comment]
@andy-io according to the dump I posted above, it's in autobahn/asyncio/websocket.py

**Comment by oberstet** [comment]
> Arguably, this is a design flaw in autobahn & txaio, but it does affect the dependent code in bitex. 

No, the reason is: a) you cannot run 2 event loops in parallel (Twisted and asyncio), and b) Twisted, once a reactor has been installed, never returns from the event loop, other than exiting the program - cannot use or installed a different reactor.

**Comment by emestee** [comment]
@oberstet no, the design flaw is that autobahn doesn't let the library consumer to choose the reactor but simply asserts that it wants to use a given one. If another library then does the same with a different reactor, there's a conflict.

**Comment by oberstet** [comment]
@emestee no, you don't understand what is going on: just don't call `choosereactor`, and Autobahn won't install one. https://github.com/crossbario/autobahn-python/blob/master/autobahn/twisted/choosereactor.py

**Comment by emestee** [comment]
@oberstet *I* don't call `choosereactor`, the autoloaded Poloniex integration does implicitly, which is what this ticket was about
